File: advance-algorithms-functions.php
//Bubble sort

class BubbleSort {
    public static function sort(array &$arr) {
        $n = count($arr);

        for ($i = 0; $i < $n - 1; $i++) {
            for ($j = 0; $j < $n - $i - 1; $j++) {
                if ($arr[$j] > $arr[$j + 1]) {
                    // Swap $arr[$j] and $arr[$j + 1]
                    $temp = $arr[$j];
                    $arr[$j] = $arr[$j + 1];
                    $arr[$j + 1] = $temp;
                }
            }
        }
    }
}

// Example usage:
$numbers = [64, 34, 25, 12, 22, 11, 90];
BubbleSort::sort($num);

echo "Sorted array: " . implode(", ", $num);

File: advance-algorithms-functions.py
# Shellsort algorithm

def shell_sort(arr):
    n = len(arr)
    gap = n // 2

    while gap > 0:
        for i in range(gap, n):
            temp = arr[i]
            j = i
            while j >= gap and arr[j - gap] > temp:
                arr[j] = arr[j - gap]
                j -= gap
            arr[j] = temp
        gap //= 2

# Example:
my_list = [12, 34, 54, 2, 3]
shell_sort(my_list)
print("Sorted array:", my_list)

